_format/1.1/index.md

```diff
@@ -536,9 +536,10 @@ Within this object, a link **MUST** be represented as either:
     link.
   * Any link-specific target attributes described below.
 
-Except for the `profile` key, each key present in a links object **MUST** have
-a single link as its value. The `profile` key, if present, **MUST** hold an
-array of links.
+Except for the `profile` key in the top-level links object and the `type` 
+key in an [error object]'s links object, each key present in a links object 
+**MUST** have a single link as its value. The aforementioned `profile` and
+`type` keys, if present, **MUST** hold an array of links.
```

Comment:
This phrasing has gotten a bit awkward, and is not entirely consistent with error `type` (which can contain a singular link or an array). I'd prefer defining `profile`'s requirements separately and then using general wording such as "Except as explicitly noted elsewhere in this specification, each key present in a links object **MUST** have a single link as its value."

Reply:
Thanks for catching this! I totally didn't realize that I was saying `type` could be a single link in one place, but that it had to be an array in another place.

If we allow `type` to be a singular link or an array, we end with two formats that clients have to support (array and object), which is more total complexity than just having one format, for no real benefit. So I'd prefer requiring `type` to be an array since, of course, a server can always send an array of one item.

Reply:
Yeah, that's fine by me - and is consistent with `profile`. I suppose when we allow an array, we should just require it for simplicity.

Just to reiterate: I really don't want all new links we introduce to be arrays by default - sometimes there's value in a single canonical link (like `self`)

Reply:
> Yeah, that's fine by me - and is consistent with `profile`.

Awesome.

> I suppose when we allow an array, we should just require it for simplicity... Just to reiterate: I really don't want all new links we introduce to be arrays by default - sometimes there's value in a single canonical link (like self)

Yup, agreed. That is, if array is allowed, it's required, but there could be links where an array isn't allowed (like `self`). 

I think the number of link types that are truly singular is pretty small, tbh, and, if we were starting from scratch, I'd say to have only one format (arrays). Then, we'd enforce that certain link types (`self`) can only hold one link through a normative MUST. But that ship has sailed, and there's (probably) not an elegant, backwards-compatible way to get there.